smdebug/core/utils.py

```diff
@@ -590,3 +591,52 @@ def check_smdataparallel_env():
             _smdataparallel_imported = None
 
     return _is_invoked_via_smddp
+
+
+def check_smmodelparallel_training():
+    """
+    The function checks whether the current job is using model parallel strategy.
+    For the training job that uses smmodelparallel, SageMaker sets following environment variables.
+    SM_HPS=
+    {
+        "mp_parameters": {
+            "ddp": true,
+            "microbatches": 4,
+            "optimize": "speed",
+            "partitions": 2,
+            "pipeline": "interleaved",
+            "placement_strategy": "spread"
+        }
+    }
+    The 'partitions' variable is a required parameter for scheduling a model parallel training job.
+
+    :return: True or False
+    """
+    global _is_using_smmodelparallel
+    if _is_using_smmodelparallel is not None:
+        return _is_using_smmodelparallel
+    if os.getenv("SM_HPS") is None:
+        _is_using_smmodelparallel = False
+    else:
+        try:
+            smp_flag = json.loads(os.getenv("SM_HPS"))
```

Comment:
Why are we using this to determine whether this a model parallel job or not? Why are not simply checking:

```
if _smp_imported and _smp_imported.core.initialized:
```

This is what `get_distributed_worker` already does. This is a stronger check because we would be referencing the model parallel package directly for the flag. 

https://github.com/awslabs/sagemaker-debugger/blob/master/smdebug/core/utils.py#L424

Reply:
discussed offline.